Text Javadoc could be improved

`Text` Javadoc does not describe the component well enough. Even though `Text` extends `Component`, it is unclear that it can't be styled. An attempt to style a `Text` also results in an unclear error message:
>Caused by: java.lang.UnsupportedOperationException: This instance is immutable

Javadoc should provide more information and probably an `@see` to `Span`.

--- a/flow-server/src/main/java/com/vaadin/flow/component/Text.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/Text.java
@@ -19,6 +19,17 @@ import com.vaadin.flow.dom.Element;
 
 /**
  * A component which encapsulates the given text in a text node.
+ * <p>
+ * Text node doesn't support setting any attribute or property so you may not
+ * use Element API (and {@link Text} doesn't provide any such contract) for
+ * setting attribute/property. It implies that you may not style this component
+ * as well. Any attempt to set attribute/property value throws an exception. The
+ * only available API for a {@link Text} component is set the text.
+ * <p>
+ * If you need a text component which can be styled then check {@code Span}
+ * class (from {@code flow-html-components}) module.
+ *
+ * @see {@code span} HTML element
  *
  * @author Vaadin Ltd
  * @since 1.0
@@ -67,6 +78,25 @@ public class Text extends Component implements HasText {
                 + "represent an HTML Element but a text Node on the client side.");
     }
 
+    /**
+     * The method is not supported for the {@link Text} class.
+     * <p>
+     * Always throws an {@link UnsupportedOperationException}.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public void setId(String id) {
+        super.setId(id);
+    }
+
+    /**
+     * The method is not supported for the {@link Text} class.
+     * <p>
+     * Always throws an {@link UnsupportedOperationException}.
+     *
+     * @throws UnsupportedOperationException
+     */
     @Override
     public void setVisible(boolean visible) {
         throw new UnsupportedOperationException("Cannot change "

--- a/flow-server/src/main/java/com/vaadin/flow/component/Text.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/Text.java
@@ -24,12 +24,10 @@ import com.vaadin.flow.dom.Element;
  * use Element API (and {@link Text} doesn't provide any such contract) for
  * setting attribute/property. It implies that you may not style this component
  * as well. Any attempt to set attribute/property value throws an exception. The
- * only available API for a {@link Text} component is set the text.
+ * only available API for a {@link Text} component is set a text.
  * <p>
  * If you need a text component which can be styled then check {@code Span}
  * class (from {@code flow-html-components}) module.
- *
- * @see {@code span} HTML element
  *
  * @author Vaadin Ltd
  * @since 1.0

